projects/storefrontlib/cms-structure/guards/multi-cms-page-guard.service.ts

```diff
@@ -0,0 +1,40 @@
+/*
+ * SPDX-FileCopyrightText: 2024 SAP Spartacus team <spartacus-team@sap.com>
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+import { Injectable, inject } from '@angular/core';
+import { CX_PAGE_GUARD_TOKEN } from './cms-page-guard.provider';
+import { RouterStateSnapshot, UrlTree } from '@angular/router';
+import { Observable, concat, endWith, first, of, skipWhile } from 'rxjs';
+import { CmsGuardsService } from '../services';
+import { CmsActivatedRouteSnapshot, UnifiedInjector } from '@spartacus/core';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class MultiCmsPageGuardService {
+  protected cxPageGuards = inject(CX_PAGE_GUARD_TOKEN);
```

Comment:
Let me think a bit about the naming for the token and accompanying service(s) in the core mechanism and I'll get back to you...

Reply:
done. proposed changes in another PR do develop https://github.com/SAP/spartacus/pull/18781